### DISCUSSION TITLE
Problema ao Retorna lista detalhada dos produtos cadastrados do arquivo JSON via Mockoon na página WEB |  Criando Apps Web com Angular e UiKit

### DISCUSSION THREAD

**Discussion by lxxzh** [balta-io/forum#160] [Tira Dúvidas] [answered]
Boa noite, estou tentando Retorna uma lista detalhada de todos produtos cadastrados do arquivo JSON via Mockoon no products-page.component.html, mas ao exibir no HTML, o mesmo não mostra as IMAGES, TITLE e PRICE, pois fica em branco. Segue as páginas abaixo.  @andrebaltieri 

<img width="948" alt="Captura de tela_20221213_005002" src="https://user-images.githubusercontent.com/106619427/207229521-3d6f3196-d547-43ad-a5bd-dbdc66a22c3e.png">
<img width="958" alt="Captura de tela_20221213_004949" src="https://user-images.githubusercontent.com/106619427/207229524-09fae32f-08fa-43e5-9de1-d6ecd871ad78.png">
<img width="958" alt="Captura de tela_20221213_005339" src="https://user-images.githubusercontent.com/106619427/207229880-87118645-ac28-491e-bf35-372385034fbb.png">
<img width="959" alt="Captura de tela_20221213_005503" src="https://user-images.githubusercontent.com/106619427/207230026-1c4982a4-4312-439a-ba67-ff54bddcdebe.png">
<img width="957" alt="Captura de tela_20221213_004934" src="https://user-images.githubusercontent.com/106619427/207229529-8115f616-a197-434e-8d5e-af83d6baa548.png">
<img width="958" alt="Captura de tela_20221213_004831" src="https://user-images.githubusercontent.com/106619427/207229532-fc29748f-f8c1-406b-b871-f22675cd2eb8.png">

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @lxxzh existe algum erro no console?

**Reply by lxxzh** [reply]
Bom dia @andrebaltieri , nenhum erro no console.

<img width="957" alt="Captura de tela_20221213_092954" src="https://user-images.githubusercontent.com/106619427/207334932-dc913c7b-35fd-4d2d-9031-9e47c79f7c23.png">

**Reply by andrebaltieri** [reply]
Boa tarde, @lxxzh, que estranho... está usando as mesmas versões do curso?
Tentou renderizar algum HTML mais simples para ver se não é algo da semânica?

Tipo assim:
```html
<ul *ngFor="let product of products$ | async">
  <li>{{ product.title }}</li>
</ul>
```

**Reply by lxxzh** [reply]
Tarde @andrebaltieri, Está na mesma versão do curso, pq até então todo passo a passo até aqui eu consegui acompanhar e desenvolver o APP.

E Fazendo um teste da forma que você passou apresentou os titulos na página. Conforme print abaixo.

<img width="954" alt="Captura de tela_20221213_171017" src="https://user-images.githubusercontent.com/106619427/207444301-d32d370f-5ec7-4c71-8b86-4ba28e3b3d22.png">


<img width="958" alt="image" src="https://user-images.githubusercontent.com/106619427/207444350-7405dc19-68c7-4f43-b786-e1ee562606fc.png">

- Quando você fala em semântica, é a forma que o HTML está no código?

**Comment by lxxzh** [comment]
@andrebaltieri  Após alguns testes, verifiquei que o problema estava no arquivo JSON, entendi a dinamica da aplicação e alterei os dados no Mockoon e após isso obtive êxito. O possível problema está no primeiro produto de "products" do arquivo JSON, (código estava quebrado), eu só copiei os produtos debaixo e alterei o title e a images e deu certo. 

<img width="957" alt="image" src="https://user-images.githubusercontent.com/106619427/207496033-36f9bce7-e3b6-4a94-8290-aedfb8634780.png">

**Reply by andrebaltieri** [reply]
Wowww @lxxzh muito bom 💜
Se puder, detalha melhor o erro aí... pode ajudar outras pessoas!!!

**Reply by moisesalves1** [reply]
Enfrentei o mesmo problema e identifique a falha no primeiro produto. Segue:

- Original
![image](https://user-images.githubusercontent.com/87319078/219963331-0fa5333e-376e-4b62-8e78-369dd906eb2e.png)

- Após alteração
![image](https://user-images.githubusercontent.com/87319078/219963389-6d00d235-08ad-45ed-8a1a-99b2e6c8516b.png)

Após correção os dados foram exibidos,, mas gostaria de informar que as imagens não estão disponíveis no blob
![image](https://user-images.githubusercontent.com/87319078/219963674-620fe161-95c6-46c8-8c64-0737b199aca4.png)

**Reply by andrebaltieri** [reply]
@moisesalves1 muito obrigado! Poderia disponibilizar seu arquivo JSON?

**Reply by moisesalves1** [reply]
Segue 
[7187_2023-02-20.zip](https://github.com/balta-io/forum/files/10788007/7187_2023-02-20.zip)

Alterei os endereços de imagem para um placekitten
